roles/privacyidea_ha/tasks/prepare.yml

```diff
@@ -0,0 +1,51 @@
+---
+# Install packages, define variables and groups, basic ufw configuration and run checks
+
+- name: Install packages
+  apt:
+    name: "{{ privacyidea_packages }}"
+    state: present
+    update_cache: yes
+
+- name: Set unique name of ha cluster (used for grouping)
+  set_fact:
+    unique_ha_group: "{{cluster_name }}_{{ project }}_{{ stage }}"
+
+- name: Create ansible groups for the cluster
+  changed_when: false
+  group_by:
+    key: "{{ unique_ha_group }}"
+
+- name: Determine if the host is a main node (first in group)
+  set_fact:
+    is_main_node: "{{ (groups[unique_ha_group] | sort | first) == inventory_hostname }}"
+
+- name: Determine internal IP
+  set_fact:
+    internal_ip: "{{ ansible_all_ipv4_addresses | select('match', '10\\..*') | first }}"
+
+# This requires that the domain of the nodes start with <name>-<index>. and the main domain with <name>.
+# Alternative: Get IP from terraform output of a defined resource pool
```

Comment:
I think it would be better to get it from an Ansible variable (that could in turn get its value from the Terraform output)

Reply:
Can't be done easily as the servers don't have a Terraform run